server/controllers/cohort.js

```diff
@@ -488,15 +488,64 @@ const addMember = (req, resp) => {
     });
 };
 
+const leaveCohort = (req, resp) => {
+  const { id: cohortId } = req.params;
+  const { userId } = req.body;
+  const sanitizedUserId = sanitize(userId);
+  const sanitizedCohortId = sanitize(cohortId);
+
+  Cohort.findOne({
+    _id: sanitizedCohortId,
+    memberIds: { $in: [sanitizedUserId] }
+  })
+    .exec()
+    .then(doc => {
+      if (!doc) {
+        throw new BadRequestException();
+      }
+
+      const { memberIds, ownerIds } = doc;
+      const isOwner = checkIfArrayContainsUserId(ownerIds, userId);
+      const isMember = checkIfArrayContainsUserId(memberIds, userId);
+
+      if (ownerIds.length === 1 && isOwner) {
+        throw new BadRequestException(
+          'cohort.actions.leave-cohort-only-one-owner'
+        );
+      }
+
+      if (isOwner) {
+        ownerIds.splice(doc.ownerIds.indexOf(userId), 1);
+      }
+
+      if (isMember) {
```

Comment:
If you are owner then you still can be member? If not then you could use just `else if` block.

Reply:
Every higher role inherits from the previous one, so even when I'm an owner I'm still a member.